libsolidity/parsing/Parser.cpp

```diff
@@ -1051,6 +1052,8 @@ ASTPointer<ForStatement> Parser::parseForStatement(ASTPointer<ASTString> const&
 
 ASTPointer<EmitStatement> Parser::parseEmitStatement(ASTPointer<ASTString> const& _docString)
 {
+	solAssert(m_scanner->currentToken() == Token::Emit, "Expected emit token.");
```

Comment:
This should use `expectToken`.

Reply:
@axic, @chriseth requested an assertion here, but using ``expectToken`` seems to be more consistent with the rest of the parser. I'd be in favor of rather reporting a fatal parser error than raising an interal compiler error.

Reply:
I thought that it would be invalid use of the function to call it when there is no emit token, but yeah, expectToken is probably still better.